src/csv/lexer.cr

```diff
@@ -89,15 +89,15 @@ abstract class CSV::Lexer
       when '\0'
         raise "unclosed quote"
         break
-      when '"'
+      when @quote_char
         case next_char
-        when ','
+        when @separator
           check_last_empty_column
           break
         when '\r', '\n', '\0'
           break
-        when '"'
-          @buffer << '"'
+        when @quote_char
```

Comment:
@asterite quick question here about the usage of instance variables. Since `@quote_char` has no setter from the outside of the CSV instance, is safe to assume the instance variable will not change?